core/serialization/blocks.ts

```diff
@@ -86,20 +86,21 @@ export function save(
     addInputBlocks = true,
     addNextBlocks = true,
     doFullSerialization = true,
+    saveIds = true,
   }: {
     addCoordinates?: boolean;
     addInputBlocks?: boolean;
     addNextBlocks?: boolean;
     doFullSerialization?: boolean;
+    saveIds?: boolean;
   } = {},
 ): State | null {
   if (block.isInsertionMarker()) {
     return null;
   }
-
   const state = {
     'type': block.type,
-    'id': block.id,
+    'id': saveIds ? block.id : null,
```

Comment:
We actually want this to be `undefined` instead of `null`. `null` gets stringified and takes up space in the save data, but `undefined` does not get serialized.

E.g. if you pass an object like the following to `JSON.stringify`:
```
{
  propertyA: null,
  propertyB: undefined
}
```
You get:
```
'{"propertyA": null}'
```

Reply:
Got it, I hadn't thought of it that way